src/modules/Hooks.ts

```diff
@@ -570,4 +580,14 @@ export class Hooks {
 			new CustomEvent<HookEventDetail>(`swup:${hook}`, { detail, bubbles: true })
 		);
 	}
+
+	/**
+	 * Parse a hook name into the name and any modifiers.
+	 * @param hook Name of the hook.
+	 */
+	parseName(hook: HookName | HookNameWithModifier): [HookName, Partial<HookOptions>] {
+		const [name, ...modifiers] = hook.split('.');
+		const options = modifiers.reduce((acc, mod) => ({ ...acc, [mod]: true }), {});
```

Comment:
This line is too cryptic for my taste. I'd prefer something like this:

```js
const options = {} as { [key: string]: boolean };
modifiers.forEach((modifier) => {
	options[modifier] = true;
});
```

Reply:
Keeping for simple type ergonomics.